Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -616,6 +621,18 @@ public class GraphTraverser: GraphTraversing {
         return Set(dependencies)
     }
 
+    public func executableDependencies(path: Path.AbsolutePath, targetName: String) -> Set<GraphDependencyReference> {
```

Comment:
Is this extra method necessary or can the `.app` product be included as part of the return value of [copyProductDependencies](https://github.com/tuist/tuist/blob/4d879469bdc81d1a5b962d22d28c4b3126a2a180/Sources/TuistCore/Graph/GraphTraverser.swift#L602)?

Also, should we include the executables in the build phase only when the target is of type of UI or unit test? I don't think it'd make sense to copy the `.app` target to `.framework`, would it?

Reply:
1 -> I'll remove it and add to `copyProductDependencies` :)
2 -> I think it can also be useful in other cases (still want to make sure, but maybe also solves this: https://github.com/tuist/tuist/issues/6130)
In general, there are two approaches that can be taken here:

A) tailor the solution for for `.app` dependencies in a `unit/uiTest` targets
B) use a more general solution, where general dependencies on apps from another project will be supported

The thinking with the latter is that it will probably cover more cases where an app from another target is used as a dependency, but maybe we rather add each of these cases individually to make it safer. What do you think?

Reply:
> The thinking with the latter is that it will probably cover more cases where an app from another target is used as a dependency, but maybe we rather add each of these cases individually to make it safer. What do you think?

I'd lean to A) unless the changes here also solve https://github.com/tuist/tuist/issues/6130. It's usually better to minimize the behavior change to what one is trying to fix. It's often difficult to foresee all the implications of updating foundational logic in the generation process.

Reply:
That being said, would you mind checking if we can also fix https://github.com/tuist/tuist/issues/6734 with some minimal changes in this PR? That issue seems quite related.

Reply:
So:
1. Re https://github.com/tuist/tuist/issues/6130 -> the app in the example project doesn't reference the watch app, but since it's an `.app` it should be the same, and indeed - after the change in this PR, the watchapp is embedded as expected
2. Re https://github.com/tuist/tuist/issues/6734 -> it's pretty similar indeed, here there is an appExtension embedding phase (`generateAppExtensionsBuildPhase`), but it calls `graphTraverser.appExtensionDependencies`, which only considers local dependencies. So I also added the appExtension references in the graph traverser, and changed `extensionKitExtensionDependencies` to look at all dependencies. It's probably the same for the graph traverser's `extensionKitExtensionDependencies` and `appClipDependencies`, but for the sake of targeted changes - only changed `appExtensionDependencies` for now.
For ease of review, added all the changes in a new commit (to be squashed later)